umpost/um2netcdf.py

```diff
@@ -92,46 +102,194 @@ def convert_proleptic(time):
+    dlat: (float) meridional spacing between latitude grid points.
+    NB - Only applies to variables on the main horizontal grids, 
+    and not the river grid.
+    """
+
+    if lat_coordinate.name() != LAT_COORD_NAME:
+        raise ValueError(
+                f"Wrong coordinate {lat_coordinate.name()} supplied. "
+                f"Expected {LAT_COORD_NAME}."
+            )
+
+    if is_lat_river_grid(lat_coordinate.points):
+        lat_coordinate.var_name = 'lat_river'
+    elif is_lat_v_grid(lat_coordinate.points, grid_type, dlat):
+        lat_coordinate.var_name = 'lat_v'
+    else:
+        lat_coordinate.var_name = 'lat'
+
+def fix_lon_coord_name(lon_coordinate, grid_type, dlon):
+    """
+    Add a 'var_name' attribute to a longitude coordinate object
+    based on the grid it lies on.
+
+    Parameters
+    ----------
+    lon_coordinate: coordinate object from iris cube (edits in place).
+    grid_type: (string) model horizontal grid type.
+    dlon: (float) zonal spacing between longitude grid points. 
+    NB - Only applies to variables on the main horizontal grids, 
+    and not the river grid.
+    """
+
+    if lon_coordinate.name() != LON_COORD_NAME:
+        raise ValueError(
+                f"Wrong coordinate {lon_coordinate.name()} supplied. "
+                f"Expected {LAT_COORD_NAME}."
+            )
+    
+    if is_lon_river_grid(lon_coordinate.points):
+        lon_coordinate.var_name = 'lon_river'
+    elif is_lon_u_grid(lon_coordinate.points, grid_type, dlon):
+        lon_coordinate.var_name = 'lon_u'
     else:
-        lat.var_name = 'lat'
-
-    lon = cube.coord('longitude')
-    if len(lon.points) == 360:
-        lon.var_name = 'lon_river'
-    elif (lon.points[0] == 0 and grid_type == 'EG') or \
-         (np.allclose(0.5*dlon, lon.points[0]) and grid_type == 'ND'):
-        lon.var_name = 'lon_u'
+        lon_coordinate.var_name = 'lon'
+
+
+def is_lat_river_grid(latitude_points):
+    """
+    Check whether latitude points are on the river routing grid.
+
+    Parameters
+    ----------
+    latitude_points: (array) 1D array of latitude grid points.
+    """
+    return len(latitude_points) == 180
+
+
+def is_lon_river_grid(longitude_points):
+    """
+    Check whether longitude points are on the river routing grid.
+
+    Parameters
+    ----------
+    longitude_points: (array) 1D array of longitude grid points.
+    """
+
+    return len(longitude_points) == 360
+
+
+def is_lat_v_grid(latitude_points, grid_type, dlat):
+    """
+    Check whether latitude points are on the lat_v grid.
+
+    Parameters
+    ----------
+    latitude_points: (array) 1D array of latitude grid points.
+    grid_type: (string) model horizontal grid type.
+    dlat: (float) meridional spacing between latitude grid points.
+    """
+    
+    min_latitude = latitude_points[0]
+    min_latitude_v_ND = -90.+0.5*dlat
+    return (
+        (min_latitude == -90 and grid_type == GRID_END_GAME) or
```

Comment:
`-90` could probably be converted to a constant to reduce magic numbers in the func.

Reply:
Good idea. I've had a go at this but unsure of the best approach.

In the condition `min_latitude == -90`, the `-90` refers to the minimum latitude in the `end game` grid type's `latitude_v` grid, and so it would make sense to define `min_lat_v_eg_grid = -90`. 

For me, I think it reads clearest if we define this at the same time we define the minimum `lat_v` for the `new dynamics` grid `min_lat_v_nd_grid`. This has to be set within the function as it depends on the `dlat` argument, and so I've tried:

```python
    min_latitude = latitude_points[0]
    min_lat_v_nd_grid = -90.+0.5*dlat
    min_lat_v_eg_grid = -90
    return (
        (min_latitude == min_lat_v_eg_grid and grid_type == GRID_END_GAME) or
```
rather than setting it as a global constant. Let me know if this looks ok!

Reply:
This naming works equally as well as module level constants, given the vars still get descriptive names.

I also noticed the `return (...)` block is trickier to parse, with an if block being simpler:
```Python
...
min_lat_v_eg_grid = -90

if grid_type == GRID_END_GAME:
    return min_latitude == min_lat_v_eg_grid
elif grid_type == GRID_NEW_DYNAMICS:
    return np.allclose(min_lat_v_nd_grid, min_latitude)
```

Reply:
Yeah great idea, that makes it easier to read/understand. I've added this in.